README.md

```diff
@@ -214,7 +214,8 @@ fn main() {
 }
 ```
 
-Tilde/Caret version requirements (like `^1.0` or `~1.2`) can be specified as well.
+You can also omit the patch version when specifying the MSRV, so `msrv = 1.30`
+is equivalent to `msrv = 1.30.0`.
```

Comment:
@flip1995 I think omitting the patch version might not work. Parsing `1.30` doesn't throw an error but, comparisons with the Lint's MSRV are not correct. Specifying the MSRV as `^1.30` works though

I just ran the snippet  below

```rust
use semver::{Version, VersionReq};

fn main() {
  const MANUAL_STRIP_MSRV: Version = Version {
      major: 1,
      minor: 45,
      patch: 0,
      pre: Vec::new(),
      build: Vec::new(),
  };
  match VersionReq::parse("1.30") {
    Ok(vReq) => {
      if !vReq.matches(&MANUAL_STRIP_MSRV) {
        println!("Will lint!")
      } else {
        println!("Won't lint!")
      }
    },
    Err(_) => println!("Error")
  }
}
```

Reply:
After playing around a bit, I think, that VersionReq is broken :sweat_smile: Look at this: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f5f25e18aa79eaa8a6837773c6ef1561)

Not what I would expect for the output. I think @taiki-e is right and we should write our own version parser.

Reply:
@flip1995 shall we also open an issue on `semver`?

Reply:
Yeah, can you please do this. I'm currently writing a `RustVersion` crate.

Reply:
Done: https://crates.io/crates/rustc-semver

Reply:
>After playing around a bit, I think, that VersionReq is broken

Uhm, the second result seems indeed weird, should be `>=1.1.0, <2.0.0` like with the caret, right? It seems to behave as if a tilde was used.

---

I'm thinking, wouldn't be enough to use `semver::Version` for both the MSRV and the version which introduces the suggestion, and compare them with `<=`, like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ebb379cc716b1951dd05b2fc7825a4a3)?

Reply:
@ebroto using `Version` for both is what we did initially but then thought that being able to ignore the patch version would be a nice to have since major changes aren't introduced in patches

I've opened an issue here: https://github.com/steveklabnik/semver/issues/221, maybe you guys can add your observations?

Reply:
> 
> 
> Done: https://crates.io/crates/rustc-semver

@flip1995 I can work on switching to this new crate instead of using `semver`

Reply:
> @flip1995 I can work on switching to this new crate instead of using `semver`

That would be great. I want to get this crate to 100% code coverage though. This will be my weekend project.

Reply:
@Suyash458 [`rustc-semver`](https://crates.io/crates/rustc-semver) now has 100% code coverage (tested in CI) and has a 1.0.0 release.

Reply:
> 
> 
> @Suyash458 [`rustc-semver`](https://crates.io/crates/rustc-semver) now has 100% code coverage (tested in CI) and has a 1.0.0 release.

And it's not even the weekend yet 😄